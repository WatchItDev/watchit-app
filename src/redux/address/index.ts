import { createSlice, PayloadAction } from '@reduxjs/toolkit';

interface AddressState {
  address: string;
  profileId: string;
  showRainbow: boolean;
}

const initialState: AddressState = {
  address: '',
  profileId: '',
  showRainbow: false,
};

const addressSlice = createSlice({
  name: 'address',
  initialState,
  reducers: {
    storeAddress(state, action: PayloadAction<{ address: string; profileId: string }>) {
      state.address = action.payload.address;
      state.profileId = action.payload.profileId;
    },

    toggleRainbow(state) {
      state.showRainbow = !state.showRainbow;
    },

  },
});

export const { storeAddress, toggleRainbow } = addressSlice.actions;
export default addressSlice.reducer;
