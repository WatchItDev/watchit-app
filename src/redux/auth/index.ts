import {createSlice, PayloadAction} from "@reduxjs/toolkit";

export interface AuthReducerState {
  session: any;
  isSessionLoading: boolean;
  isLoginModalOpen: boolean;
  balance: number;
  currentStep: number;
  isUpdatingMetadata: boolean;
  email: string;
}

const initialState: AuthReducerState = {
  session: null,
  isSessionLoading: false,
  isLoginModalOpen: false,
  balance: 0,
  currentStep: 0,
  isUpdatingMetadata: false,
  email: "",
};

const authSlice = createSlice({
  name: "auth",
  initialState,
  reducers: {
    openLoginModal: (state) => {
      state.isLoginModalOpen = true;
    },
    closeLoginModal: (state) => {
      state.isLoginModalOpen = false;
    },
    setAuthLoading: (state, action: PayloadAction<Pick<AuthReducerState, "isSessionLoading">>) => {
      state.isSessionLoading = action.payload.isSessionLoading;
    },
    setBalance: (state, action: PayloadAction<Pick<AuthReducerState, "balance">>) => {
      state.balance = action.payload.balance;
    },
    setProfileCreationStep: (state, action: PayloadAction<{step: number}>) => {
      state.currentStep = action.payload.step;
    },
    resetCurrentStep: (state) => {
      state.currentStep = 0;
    },
    setSession: (state, action: PayloadAction<{session: any}>) => {
      state.session = action.payload.session;
    },
    setEmail: (state, action: PayloadAction<string>) => {
      state.email = action.payload;
    },
    updateProfileData: (
      state,
      action: PayloadAction<{
        name: string;
        bio: string;
        profileImage?: string;
        backgroundImage?: string;
      }>,
    ) => {
      if (state.session && state.session.profile) {
        state.session.profile.metadata = {
          ...state.session.profile.metadata,
          displayName: action.payload.name,
          bio: action.payload.bio,
          picture: {
            optimized: {
              uri: action.payload.profileImage ? action.payload.profileImage : undefined,
            },
          },
          coverPicture: {
            optimized: {
              uri: action.payload.backgroundImage ? action.payload.backgroundImage : undefined,
            },
          },
        };
      }
    },
    setIsUpdatingMetadata: (state, action: PayloadAction<boolean>) => {
      state.isUpdatingMetadata = action.payload;
    },
  },
});

export const {
  openLoginModal,
  closeLoginModal,
  setAuthLoading,
  setBalance,
  setProfileCreationStep,
  resetCurrentStep,
  setSession,
  updateProfileData,
  setIsUpdatingMetadata,
  setEmail,
} = authSlice.actions;

export default authSlice.reducer;
