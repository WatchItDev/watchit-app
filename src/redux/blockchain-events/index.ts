import { createSlice, PayloadAction } from '@reduxjs/toolkit'
import { Log } from 'viem'

interface BlockchainEventsState {
  events: Log[];
}

const initialState: BlockchainEventsState = {
  events: [],
}

const blockchainEventsSlice = createSlice({
  name: 'blockchainEvents',
  initialState,
  reducers: {
    addBlockchainEvent(state, action: PayloadAction<Log>) {
      state.events.push(action.payload)
    },
    setBlockchainEvents(state, action: PayloadAction<Log[]>) {
      state.events = action.payload
    },
    clearBlockchainEvents(state) {
      state.events = []
    },
  },
})

export const { addBlockchainEvent, setBlockchainEvents, clearBlockchainEvents } =
  blockchainEventsSlice.actions

export default blockchainEventsSlice.reducer
