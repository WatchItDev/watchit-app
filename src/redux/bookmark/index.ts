import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { Post } from '@src/graphql/generated/graphql.ts';

export interface BookmarkReducerState {
  bookmarkPublications: Post[];
  hiddenBookmarks: Post[];
}

const initialState: BookmarkReducerState = {
  bookmarkPublications: [],
  hiddenBookmarks: [],
};

const bookmarkSlice = createSlice({
  name: 'bookmark',
  initialState,
  reducers: {
    // Add a post to bookmarks and remove it from hidden if it is
    addBookmark: (state, action: PayloadAction<Post>) => {
      const isAlreadyBookmarked = state.bookmarkPublications.some(
        (publication) => publication.id === action.payload.id
      );

      if (!isAlreadyBookmarked) {
        state.bookmarkPublications.push(action.payload);
        // Remove from hiddenBookmarks if exists
        state.hiddenBookmarks = state.hiddenBookmarks.filter(
          (publication) => publication.id !== action.payload.id
        );
      }
    },

    // Move a post to hiddenBookmarks instead of deleting it completely
    removeBookmark: (state, action: PayloadAction<string>) => {
      const publicationToHide = state.bookmarkPublications.find(
        (publication) => publication.id === action.payload
      );

      if (publicationToHide) {
        state.hiddenBookmarks.push(publicationToHide);
        state.bookmarkPublications = state.bookmarkPublications.filter(
          (publication) => publication.id !== action.payload
        );
      }
    },

    // Remove all bookmarks
    clearBookmarks: (state) => {
      state.bookmarkPublications = [];
    },
  },
});

export const { addBookmark, removeBookmark, clearBookmarks } = bookmarkSlice.actions;

export default bookmarkSlice.reducer;
