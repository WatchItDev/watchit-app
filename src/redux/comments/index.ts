import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { AnyPublication } from '@lens-protocol/api-bindings';

type PendingComment = AnyPublication & { uri: string };

export type CommentsReducerState = {
  refetchTriggerByPublication: {
    [publicationId: string]: number;
  };
  hiddenComments: AnyPublication[];
  counterLikes: { [publicationId: string]: number };
  comments: { [publicationId: string]: AnyPublication[] };
  pendingComments: { [publicationId: string]: PendingComment[] };
};

const initialState: CommentsReducerState = {
  refetchTriggerByPublication: {},
  hiddenComments: [],
  counterLikes: {},
  comments: {},
  pendingComments: {},
};

const commentsSlice = createSlice({
  name: 'comments',
  initialState,
  reducers: {
    refetchCommentsByPublication: (state, action: PayloadAction<string>) => {
      const publicationId = action.payload;
      if (!state.refetchTriggerByPublication[publicationId]) {
        state.refetchTriggerByPublication[publicationId] = 1;
      } else {
        state.refetchTriggerByPublication[publicationId] += 1;
      }
    },
    hiddeComment: (state, action: PayloadAction<AnyPublication>) => {
      state.hiddenComments.push(action.payload);
    },
    setCounterLikes: (state, action: PayloadAction<{ publicationId: string; likes: number }>) => {
      state.counterLikes[action.payload.publicationId] = action.payload.likes;
    },
    incrementCounterLikes: (state, action: PayloadAction<string>) => {
      const publicationId = action.payload;
      if (state.counterLikes[publicationId] !== undefined) {
        state.counterLikes[publicationId] += 1;
      }
    },
    decrementCounterLikes: (state, action: PayloadAction<string>) => {
      const publicationId = action.payload;
      if (state.counterLikes[publicationId] !== undefined) {
        state.counterLikes[publicationId] -= 1;
      }
    },
    addPendingComment: (
      state,
      action: PayloadAction<{ publicationId: string; comment: PendingComment }>
    ) => {
      const { publicationId, comment } = action.payload;
      if (!state.pendingComments[publicationId]) {
        state.pendingComments[publicationId] = [];
      }
      // Prepend new comment to the beginning of the list
      state.pendingComments[publicationId] = [comment, ...state.pendingComments[publicationId]];
    },
    removePendingComment: (
      state,
      action: PayloadAction<{ publicationId: string; commentId: string }>
    ) => {
      const { publicationId, commentId } = action.payload;

      // Delete the comment from the pending list
      state.pendingComments[publicationId] = state.pendingComments[publicationId].filter(
        (comment) => comment.id !== commentId
      );
    },
  },
});

export const {
  hiddeComment,
  refetchCommentsByPublication,
  setCounterLikes,
  incrementCounterLikes,
  decrementCounterLikes,
  addPendingComment,
  removePendingComment,
} = commentsSlice.actions;

export default commentsSlice.reducer;
