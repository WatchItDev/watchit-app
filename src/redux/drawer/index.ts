import { createSlice } from '@reduxjs/toolkit'

const initialState = {
  open: false,
}

const drawerSlice = createSlice({
  name: 'drawer',
  initialState,
  reducers: {
    openDrawer: (state) => {
      state.open = true
    },
    closeDrawer: (state) => {
      state.open = false
    },
    toggleDrawer: (state) => {
      state.open = !state.open
    },
  },
})

export const { openDrawer, closeDrawer, toggleDrawer } = drawerSlice.actions

export default drawerSlice.reducer
