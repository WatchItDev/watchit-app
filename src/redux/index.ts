export { minibarInitialState } from "@redux/minibar";
export { drawerInitialState } from "@redux/drawer";
export { authInitialState } from "@redux/auth";
export { bookmarkInitialState } from "@redux/bookmark";
export { commentsInitialState } from "@redux/comments";
export { notificationsInitialState } from "@redux/notifications";
export { followersInitialState } from "@redux/followers";
export { addressInitialState } from "@redux/address";
export { blockchainEventsInitialState } from "@redux/blockchain-events";
export { transactionInitialState } from "@redux/transactions";
