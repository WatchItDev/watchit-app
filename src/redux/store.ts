import { type Store } from 'redux';
import { configureStore } from '@reduxjs/toolkit';
import rootReducer from '@src/redux/reducer';

export const createStore = (initialState = {}): Store => {
  return configureStore({
    reducer: rootReducer,
    devTools: true,
    preloadedState: initialState,
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware({
        serializableCheck: false,
      }),
  });
};

export type RootState = ReturnType<typeof store.getState>;

export const store = createStore();
