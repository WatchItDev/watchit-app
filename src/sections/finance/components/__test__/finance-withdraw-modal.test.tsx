import { describe, expect, it } from "vitest";
import { fireEvent, render, screen } from "@testing-library/react";
import FinanceWithdrawModal from "../finance-withdraw-modal";
import { Provider } from "react-redux";
import { store } from "@src/redux/store";

vi.mock("@src/hooks/use-auth.ts", () => ({
  useAuth: vi.fn(() => ({
    session: {
      address: "0x1111111111111111111111111111111111111111",
    },
  })),
}));

const mockWithdraw = vi.fn();

vi.mock("@src/workers/backgroundTaskWorker?worker", () => {
  return {
    default: class {
      postMessage() {}
      terminate() {}
      addEventListener() {}
      removeEventListener() {}
    },
  };
});

vi.mock("@src/hooks/protocol/use-withdraw.ts", () => ({
  useWithdraw: () => ({
    withdraw: mockWithdraw,
    isLoading: false,
    error: null,
  }),
}));

vi.mock("@src/hooks/protocol/use-get-vault-balance.ts", () => ({
  useGetVaultBalance: () => ({
    data: 1000n,
    isLoading: false,
    error: null,
  }),
}));
const mockOnClose = vi.fn();
const renderComponent = (open = true) =>
  render(
    <Provider store={store}>
      <FinanceWithdrawModal open={open} onClose={mockOnClose} />
    </Provider>,
  );

describe("[COMPONENTS] <FinanceWithdrawModal/> ", () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });
  it("to match snapshot", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it("should not render modal if open is false", () => {
    renderComponent(false);
    expect(screen.queryByTestId("finance-withdraw-modal")).not.toBeInTheDocument();
  });

  it("should render modal if open is true", () => {
    renderComponent();
    expect(screen.getByTestId("finance-withdraw-modal")).toBeInTheDocument();
  });

  it("should render tabs correctly", () => {
    renderComponent();
    expect(screen.getByText("Metamask")).toBeInTheDocument();
    expect(screen.getByText("Smart Account")).toBeInTheDocument();
  });

  it("should change content when switching tabs", () => {
    renderComponent();
    fireEvent.click(screen.getByText("Smart Account"));
    expect(screen.getByText("Smart Account")).toBeInTheDocument();
    expect(screen.getByTestId("FinanceWithdraw")).toBeInTheDocument();
    screen.debug();
  });
});
