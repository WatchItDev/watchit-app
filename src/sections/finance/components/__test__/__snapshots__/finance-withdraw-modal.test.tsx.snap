// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[COMPONENTS] <FinanceWithdrawModal/>  > to match snapshot 1`] = `
<div
  aria-hidden="true"
/>
`;
